server/src/main/java/org/elasticsearch/common/blobstore/BlobContainer.java

```diff
@@ -49,6 +49,22 @@
      */
     InputStream readBlob(String blobName) throws IOException;
 
+    /**
+     * Creates a new {@link InputStream} that can be used to read the given blob starting from
+     * a specific {@code position} in the blob. The {@code length} is an indication of the
+     * number of bytes that are expected to be read from the {@link InputStream}.
+     *
+     * @param blobName The name of the blob to get an {@link InputStream} for.
+     * @param position The position in the blob where the next byte will be read.
+     * @param length   An indication of the number of bytes to be read.
+     * @return The {@code InputStream} to read the blob.
+     * @throws NoSuchFileException if the blob does not exist
+     * @throws IOException         if the blob can not be read.
+     */
+    default InputStream readBlob(final String blobName, final long position, final int length) throws IOException {
+        throw new UnsupportedOperationException(); // NORELEASE
```

Comment:
Yes, sorry if I wasn't clear. I did not include support for other blob store repositories like S3 because it adds more code to review and more plumbing. As I don't expect any difficulty I've just put aside this for now.